Codecs/DAG-CBOR.md

```diff
@@ -0,0 +1,37 @@
+# [WIP] DagCBOR Spec
+
+DAG-CBOR supports the full ["IPLD Data Model v1."](../IPLD-Data-Model-v1.md)
+
+CBOR already natively supports all ["IPLD Data Model v1: Simple Types."](../IPLD-Data-Model-v1.md#simple-types)
+
+## Format
+
+The CBOR IPLD format is called DagCBOR to disambiguate it from regular CBOR.
+Most CBOR objects are valid DagCBOR. The only hard restriction is that any field
+with the tag 42 must be a valid CID.
+
+## Link Format
+
+As with all IPLD formats, DagCBOR must be able to encode merkle-links. In
+DagCBOR, links are encoded using the raw-binary (identity, NUL) multibase in a
+field with a byte-string type (major type 2), with the tag 42.
+
+(the inclusion of the multibase exists for historical reasons)
+
+## Map Key Restriction
+
+In DagCBOR, map keys must be strings (TODO: drop this? We already have
+unpathable map keys). Furthermore, map keys should avoid using `/` as this is
+unpathable (TODO: drop this? IMO, we should support path escaping out of the
+box).
+
+## Canonical DagCBOR
+
+Canonical DagCBOR should:
+
+1. Use no tags other than the CID tag (42). Other tags may be lost in
+   conversion.
+2. Should use the canonical CBOR encoding and field ordering. Other orderings
```

Comment:
There should be a section about numbers. I'd like to have specified what the right thing for cases like https://github.com/ipld/interface-ipld-format/issues/9#issuecomment-431029329 are.

Reply:
This was pulled in from the Dag-CBOR PR. Regardless, I'd like to merge this and then log an issue to make this improvement. I think there will end up being some discussion around this particular point.